### ISSUE TITLE
Unable to disable trigger_on_change on pipeline with Github source

### ISSUE THREAD

**Issue by ackjewtn** [awslabs/aws-deployment-framework#432]
We recently hit the CodePipeline limit "There can only be up to 50 actions in a pipeline stage" and decided to split the pipeline per region to solve it. The next limit we hit was the GitHub webhook limit per repository, which is 20. We then built a custom solution to trigger our pipelines, but we're unable to disable the webhook creation in the deployment_map using `trigger_on_changes: False`

We're currently on `ADF 3.1.2`

```yaml
  - name: sample-pipeline-us-east-1
    default_providers:
      source:
        provider: github
        properties:
          repository: repo-name
          owner: ourcompany
          oauth_token_path: /adf/token
          json_field: token
          trigger_on_changes: False
      build:
        provider: codebuild
        properties:
          spec_filename: sample-pipeline/buildspec.yml
          image: STANDARD_5_0
          size: medium
          environment_variables:
            CONTAINS_TRANSFORM: True
      deploy:
        provider: cloudformation
        properties:
          root_dir: sample-pipeline
          stack_name: adf-sample-stack
          action: CREATE_UPDATE
    params:
      restart_execution_on_update: True
    regions: [us-east-1]
    targets: *baseline_targets
```

When ADF runs, it fails with 
```bash
2022-01-27 13:27:44,422 \| INFO \| deployment_map \| Loading deployment_map file deployment_maps/baseline.yml \| (deployment_map.py:65)
--
110 | 2022-01-27 13:27:44,522 \| ERROR \| deployment_map \| Key 'pipelines' error:
Or({'name': And(<class 'str'>, <built-in function len>), 'default_providers': {'source': And({'provider': Or('codecommit', 'github', 's3'), 'properties': <class 'dict'>}, <function <lambda> at 0x7f28c449a9d0>), Optional('build'): And({Optional('provider'): Or('codebuild', 'jenkins'), Optional('enabled'): <class 'bool'>, Optional('properties'): <class 'dict'>}, <function <lambda> at 0x7f28c449aa60>), Optional('deploy'): And({'provider': Or('cloudformation', 's3', 'codedeploy', 'lambda', 'service_catalog', 'codebuild'), Optional('enabled'): <class 'bool'>, Optional('properties'): <class 'dict'>}, <function <lambda> at 0x7f28c449aaf0>)}, Optional('params'): {Optional('notification_endpoint'): <class 'str'>, Optional('schedule'): <class 'str'>, Optional('restart_execution_on_update'): <class 'bool'>}, Optional('tags'): <class 'dict'>, Optional('targets'): [Or(<class 'str'>, <class 'int'>, {Optional('path'): Or(<class 'str'>, <class 'int'>, [Or(<class 'str'>, <class 'int'>)]), Optional('tags'): {And(<class 'str'>, Regex('\\A.{1,128}\\Z')): And(<class 'str'>, Regex('\\A.{0,256}\\Z'))}, Optional('target'): Or(<class 'str'>, <class 'int'>, [Or(<class 'str'>, <class 'int'>)]), Optional('name'): <class 'str'>, Optional('provider'): Or('lambda', 's3', 'codedeploy', 'cloudformation', 'service_catalog', 'approval', 'codebuild', 'jenkins'), Optional('properties'): Or({Optional('image'): Or(<class 'str'>, {'repository_arn': <class 'str'>, Optional('tags'): <class 'dict'>}), Optional('size'): Or('small', 'medium', 'large'), Optional('spec_filename'): <class 'str'>, Optional('environment_variables'): {Optional(<class 'str'>): Or(<class 'str'>, <class 'bool'>, <class 'int'>, <class 'object'>)}, Optional('role'): <class 'str'>, Optional('timeout'): <class 'int'>, Optional('privileged'): <class 'bool'>, Optional('spec_inline'): <class 'object'>}, {Optional('project_name'): <class 'str'>, Optional('server_url'): <class 'str'>, Optional('provider_name'): <class 'str'>}, {Optional('stack_name'): <class 'str'>, Optional('template_filename'): <class 'str'>, Optional('root_dir'): <class 'str'>, Optional('role'): <class 'str'>, Optional('action'): Or('CHANGE_SET_EXECUTE', 'CHANGE_SET_REPLACE', 'CREATE_UPDATE', 'DELETE_ONLY', 'REPLACE_ON_FAILURE', 'change_set_execute', 'change_set_replace', 'create_update', 'delete_only', 'replace_on_failure'), Optional('outputs'): <class 'str'>, Optional('change_set_approval'): <class 'bool'>, Optional('param_overrides'): [{'inputs': <class 'str'>, 'param': <class 'str'>, 'key_name': <class 'str'>}]}, {'application_name': <class 'str'>, 'deployment_group_name': <class 'str'>, Optional('role'): <class 'str'>}, {'bucket_name': <class 'str'>, 'object_key': <class 'str'>, Optional('extract'): <class 'bool'>, Optional('role'): <class 'str'>}, {'product_id': <class 'str'>, Optional('configuration_file_path'): <class 'str'>}, {'function_name': <class 'str'>, Optional('input'): Or(<class 'str'>, <class 'object'>), Optional('role'): <class 'str'>}, {Optional('message'): <class 'str'>, Optional('notification_endpoint'): <class 'str'>, Optional('sns_topic_arn'): <class 'str'>}), Optional('regions'): Or(<class 'str'>, <class 'list'>)}, [Or(<class 'str'>, <class 'int'>)])], Optional('regions'): Or(<class 'str'>, <class 'list'>), Optional('completion_trigger'): {'pipelines': [<class 'str'>]}}) did not validate {'name': 'sample-pipeline-us-east-1, 'default_providers': {'source': {'provider': 'github', 'properties': {'repository': 'repo-name', 'owner': 'ourcompany', 'oauth_token_path': '/adf/token', 'json_field': 'token', 'trigger_on_changes': False}}, 'build': {'provider': 'codebuild', 'properties': {'spec_filename': 'sample-pipeline/buildspec.yml', 'image': 'STANDARD_5_0', 'size': 'medium', 'environment_variables': {'CONTAINS_TRANSFORM': True}}}, 'deploy': {'provider': 'cloudformation', 'properties': {'root_dir': 'sample-pipeline', 'stack_name': 'adf-sample-stack', 'action': 'CREATE_UPDATE'}}}, 'params': {'restart_execution_on_update': True}, 'regions': ['eu-north-1'], 'targets': [{'path': ['/sandbox/foo']}, {'path': ['/adf-staging', '/policy-staging', '/sandbox/team', '/sandbox/pool']}, {'path': ['/infrastructure/preprod']}, {'path': ['/infrastructure/prod', '/security/prod', '/deployment']}, {'path': ['/workloads/dev']}, {'path': ['/workloads/datadev']}, {'path': ['/workloads/ops']}, {'path': ['/workloads/preprod']}, {'path': ['/workloads/prod']}, {'path': ['/exceptions']}]}
--
112 | Key 'default_providers' error:
113 | Key 'source' error:
114 | Key 'properties' error:
115 | Wrong key 'trigger_on_changes' in {'repository': 'repo-name', 'owner': 'ourcompany', 'oauth_token_path': '/adf/token', 'json_field': 'token', 'trigger_on_changes': False} \| (deployment_map.py:73)

```

As per the documentation here: https://github.com/awslabs/aws-deployment-framework/blob/master/docs/providers-guide.md#github

```
trigger_on_changes - (Boolean) default: True.
Whether CodePipeline should release a change and trigger the pipeline. When set to False, you either need to trigger the pipeline manually, through a schedule, or through the completion of another pipeline.

This disables the triggering of changes when set to False. It will not deploy the web hook that GitHub would otherwise use to trigger the pipeline on changes.

By default, it will trigger deploy the web hook and trigger on changes using web hook call executed by GitHub.
```

Any help on this issue is much appreciated.

**Comment by ackjewtn** [comment]
Never mind, i see that the commit implementing this is in the 3.2.0 milestone...

**Comment by sbkok** [comment]
Correct, we are working on the next release that includes this feature.

For the time being, I would recommend to use the v3.1.2 release docs until the next version is ready and deployed.

**Comment by ackjewtn** [comment]
@sbkok Do you have any idea on when 3.2.0 will be released?

**Comment by sbkok** [comment]
@ackjewtn Thank you for your patience. We ran into quite a number of issues and improved ADF through refactoring.
I am happy to inform you that we [released v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
